File: regexlib/2021-5-15/javascript_test_file/regexlib_3658.js
/* 3658
 * &lt;[iI][fF][rR][aA][mM][eE]([^&gt;]*[^/&gt;]*[/&gt;]*[&gt;])
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"&lt;iframe"+"!"*5000+"! _1_POA(i)"
 */
var REGEX = new RegExp("&lt;[iI][fF][rR][aA][mM][eE]([^&gt;]*[^/&gt;]*[/&gt;]*[&gt;])")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '&lt;iframe' + '!'.repeat(i*10000) + '! _1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_3658.py
# 3658
# &lt;[iI][fF][rR][aA][mM][eE]([^&gt;]*[^/&gt;]*[/&gt;]*[&gt;])
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:"&lt;iframe"+"!"*5000+"! _1_POA(i)"

import re
from time import perf_counter

regex = """&lt;[iI][fF][rR][aA][mM][eE]([^&gt;]*[^/&gt;]*[/&gt;]*[&gt;])"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "&lt;iframe" + "!" * i * 10000 + "! _1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")